### PR TITLE
feat: add disable options to SelectArrayInput

### PR DESCRIPTION
**Pull request by paulo9mv**
Add disabled options to `SelectArrayInput` as discussed in the issue #5759

### CODE REVIEW

File: docs/Inputs.md [hunk 1]
```diff
@@ -1338,6 +1338,28 @@ const choices = [
...
+```jsx
+const choices = [
+    { _id: 123, full_name: 'Leo Tolstoi', sex: 'M' },
+    { _id: 456, full_name: 'Jane Austen', sex: 'F' },
+    { _id: 1, full_name: 'System Administrator', sex: 'F', disabled: true },
+];
+<SelectArrayInput source="author_id" choices={choices} optionText="full_name" optionValue="_id" />
+```
+
+You can use a custom field name by setting `disableValue` prop:
```
Comment (djhi) [inline docs/Inputs.md:1352]:
```suggestion
You can use a custom field name by setting the `disableValue` prop: 
```

File: docs/Inputs.md [hunk 2]
```diff
@@ -1338,6 +1338,28 @@ const choices = [
 ];
 ```
 
+You can set disabled values by setting the `disabled` property of one item:
```
Comment (djhi) [inline docs/Inputs.md:1341]:
```suggestion
You can render any item as disabled by setting its `disabled` property to `true`: 
```

**Review by djhi — CHANGES_REQUESTED** [review]
Thanks!

File: examples/demo/src/segments/data.ts [hunk 3]
```diff
@@ -4,5 +4,9 @@ export default [
     { id: 'ordered_once', name: 'resources.segments.data.ordered_once' },
     { id: 'regular', name: 'resources.segments.data.regular' },
     { id: 'returns', name: 'resources.segments.data.returns' },
-    { id: 'reviewer', name: 'resources.segments.data.reviewer' },
```
Comment (fzaninotto) [inline examples/demo/src/segments/data.ts:7]: Please revert this change, and use the simple example to illustrate the functionality. Here, it makes the e-commerce demo worse for no reason.

File: examples/demo/src/visitors/SegmentsInput.tsx [hunk 4]
```diff
@@ -17,6 +17,7 @@ const SegmentsInput: FC<Props> = ({ addField, ...rest }) => {
             choices={segments.map(segment => ({
                 id: segment.id,
                 name: translate(segment.name),
+                disabled: segment.disabled,
```
Comment (fzaninotto) [inline examples/demo/src/visitors/SegmentsInput.tsx:20]: same, please revert

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: examples/demo/src/segments/data.ts (hunk 3 above)
Comment (paulo9mv) [reply examples/demo/src/segments/data.ts]: 61f53f7

File: examples/demo/src/visitors/SegmentsInput.tsx (hunk 4 above)
Comment (paulo9mv) [reply examples/demo/src/visitors/SegmentsInput.tsx]: 61f53f7

**Comment by WiXSL** [comment]
@paulo9mv,
I rebased my PR (#5941) and no longer get this e2e failing test.
Maybe is a flaky test

**Comment by paulo9mv** [comment]
> @paulo9mv,
> I rebased my PR (#5941) and no longer got this e2e failing navigation test.
> Maybe is a flaky test

thanks @WiXSL , i will try it now

**Review by fzaninotto — APPROVED** [review]

**Comment by fzaninotto** [comment]
Thanks!
